Tests/FirebaseTests/Tests/FirebaseRequestTests.swift

```diff
@@ -0,0 +1,161 @@
+    }
+    
+    func test_firebaseRequest_should_addHeadersToRequest_when_callWithCustomHeaders() throws {
+        // Given
+        let clientStub = ClientStub(responseText: "")
+        let firebaseApiRequest = FirebaseAPIRequest.dummy(client: clientStub)
+        let fakeRoute = FakeRoute(request: firebaseApiRequest)
+        let expectedHeaderName = String.random(length: 10)
+        let expectedHeaderValue = String.random(length: 10)
+        
+        // When
+        let _ = try fakeRoute.fake(headers: [expectedHeaderName: expectedHeaderValue])
+        
+        // Then
+        guard let _ = clientStub.calls.first?.headers.contains(name: expectedHeaderName) else {
+            XCTFail("Missing expected header name")
+            return
+        }
+        XCTAssertEqual(clientStub.calls.first?.headers[expectedHeaderName].first, expectedHeaderValue)
+    }
+    
+    func test_firebaseRequest_should_addQuery_when_callWithQuery() throws {
+        // Given
+        let clientStub = ClientStub(responseText: "")
+        let firebaseApiRequest = FirebaseAPIRequest.dummy(client: clientStub)
+        let fakeRoute = FakeRoute(request: firebaseApiRequest)
+        let expectedQueryName = String.random(length: 10)
+        let expectedQueryValue = String.random(length: 10)
+        let expectedQuery = "\(expectedQueryName)=\(expectedQueryValue)"
+        
+        // When
+        let _ = try fakeRoute.fake(query: expectedQuery)
+        
+        // Then
+        XCTAssertNotNil(clientStub.calls.first?.url.query?.range(of: expectedQuery))
+    }
+    
+    func test_firebaseRequest_should_addBody_when_callWithBodyData() throws {
+        // Given
+        let clientStub = ClientStub(responseText: "")
+        let firebaseApiRequest = FirebaseAPIRequest.dummy(client: clientStub)
+        let fakeRoute = FakeRoute(request: firebaseApiRequest)
+        let expectedBody = String.random(length: 10)
+        
+        // When
+        let _ = try fakeRoute.fake(body: expectedBody)
+        
+        // Then
+        guard let call = clientStub.calls.first else {
+            XCTFail("No body found")
+            return
+        }
+        XCTAssertEqual(call.body.convertToHTTPBody().data, HTTPBody(string: expectedBody).data)
+    }
+    
+    func test_firebaseRequest_should_returnError_when_callFailed() throws {
+        // Given
+        let clientStub = ClientStub(withError: FirebaseError(error: "Error during API call"))
+        let firebaseApiRequest = FirebaseAPIRequest.dummy(client: clientStub)
+        let fakeRoute = FakeRoute(request: firebaseApiRequest)
+        let expectedBody = String.random(length: 10)
+        
+        // When
+        let result = try fakeRoute.fake(body: expectedBody)
+        
+        // Then
+        let _ = result.catch { error in
+            XCTAssertEqual(error.localizedDescription, "⚠️ [FirebaseError.Error during API call: Error during API call]")
```

Comment:
You should put the error in a variable and use it to generate this string so that a change in the error message above doesn't break the test here. 

```swift
let expectedError = FirebaseError(error: "Error during API call")
let clientStub = ClientStub(withError: expectedError)

XCTAssertEqual(error.localizedDescription, "⚠️ [FirebaseError.\(expectedError.identifier): \(expectedError.reason)]")
```

This way, it should not break if you change the definition of the `identifier` and `reason` implementation which are not the subject of this test I think. That said, you should maybe even use `expectedError.localizedDescription` if appropriate, to be fully compliant with this last comment.